core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -514,7 +514,23 @@ public void rewind(String workflowId, boolean useLatestDefinitions) {
         workflow.setStatus(WorkflowStatus.RUNNING);
         workflow.setOutput(null);
         workflow.setExternalOutputPayloadStoragePath(null);
-        executionDAOFacade.createWorkflow(workflow);
+
+        try {
+            executionDAOFacade.createWorkflow(workflow);
+        } catch (Exception e) {
+            Monitors.recordWorkflowStartError(workflowDef.getName(), WorkflowContext.get().getClientApp());
+            LOGGER.error("Unable to restart workflow: {}", workflowDef.getName(), e);
+
+            // It's possible the terminate workflow call hits an exception as well, in that case we want to log both
+            // errors to help diagnosis.
+            try {
+                terminateWorkflow(workflowId, "Error when restarting the workflow");
```

Comment:
I am not convinced that a ```try...catch``` block here is the right way to address this. The exception when terminating a workflow should be caught at a granular level within the terminate logic and this should be logged/handled accordingly.

Reply:
This exception is primarily for `executionDAOFacade.createWorkflow(workflow)`, and `terminateWorkflow` is just a clean up as part of that exception. I'd still like to throw the exception for `createWorkflow`, irrespective of what happened with `terminateWorkflow`, while logging `terminateWorkflow` exception. Thoughts?

Reply:
Yes, the exception handling for createWorkflow needs to be handled. 
My comment is regarding the extended handling for the terminate part which seems out of place here. Something like this - 


```java
try {
    executionDAOFacade.createWorkflow(workflow);
} catch (Exception e) {
    Monitors.recordWorkflowStartError(workflowDef.getName(), WorkflowContext.get().getClientApp());
    LOGGER.error("Unable to restart workflow: {}", workflowDef.getName(), e);
    terminateWorkflow(workflowId, "Error when restarting the workflow");
    throw e;
}
```